src/hotspot/os_cpu/linux_x86/os_linux_x86.cpp

```diff
@@ -459,22 +459,48 @@ bool os::supports_sse() {
 }
 
 juint os::cpu_microcode_revision() {
+  // Note: this code runs on startup, and therefore should not be slow.
+
   juint result = 0;
-  char data[2048] = {0}; // lines should fit in 2K buf
-  size_t len = sizeof(data);
-  FILE *fp = os::fopen("/proc/cpuinfo", "r");
-  if (fp) {
-    while (!feof(fp)) {
-      if (fgets(data, len, fp)) {
-        if (strstr(data, "microcode") != NULL) {
-          char* rev = strchr(data, ':');
-          if (rev != NULL) sscanf(rev + 1, "%x", &result);
-          break;
+
+  // Attempt 1 (faster): Read the microcode version off the sysfs.
+  {
+    FILE *fp = os::fopen("/sys/devices/system/cpu/cpu0/microcode/version", "r");
+    if (fp) {
+      char data[128] = {0}; // looking for short line
+      if (fgets(data, sizeof(data), fp)) {
+        sscanf(data, "%x", &result);
+      }
```

Comment:
Just use fscanf? Would not need a buffer and be more compact.

Reply:
Right, see new commit.